text/0000-pack-buildpack-create.md

```diff
@@ -0,0 +1,99 @@
+# Meta
+[meta]: #meta
+- Name: Pack Command to Create a Buildpack Repo
+- Start Date: 2021-01-19
+- Author(s): [jkutner](https://github.com/jkutner)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a new command in Pack that would create the scaffolding for a new buildpack.
+
+# Definitions
+[definitions]: #definitions
+
+- *Scaffold* - the essential files and directories required for a buildpack
+
+# Motivation
+[motivation]: #motivation
+
+Every new buildpack requires a few essential files and directories. We can make buildpack development easier by automatically generating these with a single command.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Add a new command to Pack:
+
+```
+$ pack buildpack new <id> \
+    --stack <stack> \
+    --path <path>
+```
+
+Where the flags include:
+
+* `stack` - a set of compatible stacks for the buildpack. may be specificied multiple times.
+* `path` - the location on the filesystem to generate the artifacts
+
+The command will create a directory named for the buildpack ID, or the second component of a `<namespace>/<name>` ID. Inside of that directory, it will generate the minimal artifacts required to implement a buildpack in Bash.
+
+The `pack buildpack new` is only a generator. It will not be a _living command_ (i.e. you cannot re-run it to update a buildpack).
+
+## Future Work
+
+In the future, we may add support for a `--template` flag, which would create buildpacks from templates stored in a curated repository. In this way we can support many different languages, and even support versions the same language with different opinions.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+See https://github.com/buildpacks/pack/pull/1025
+
+The only supported language will be Bash. This is based on [CNB user research](https://docs.google.com/document/d/1uNE8qkvhBCLIQUjIEbOTfT1epEt9_nHk_fNc64YPEvY/edit), which found:
+
+> Comfort with Go is required to interpret buildpacks / paketo source code
+> * We assume Go experience is not something we can expect our users to have
+
+> Part of the reason developers take to docker easily is because the commands are similar to shell scripts
+> * We assume shell scripts is what devs will be most comfortable with / wil analogize from
+
+The generated files will include:
+
+
+* `buildpack.toml`
+* `bin/build`
+* `bin/detect`
+
+The `buildpack.toml` will be pre-poluated with the values provided in the CLI command and some defaults (like `0.0.0` for the version).
```

Comment:
Would it make sense to do `0.0.1` as the default version, rather than `0.0.0`?

Reply:
i don't have a strong opinion